server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/draft/model/message/view/MessageViewFactory.java

```diff
@@ -184,37 +145,12 @@ static Instant getDateFromHeaderOrInternalDateOtherwise(Message mimeMessage, Mes
                 .orElse(message.getInternalDate().toInstant());
         }
 
-        static Instant getDateFromHeaderOrInternalDateOtherwise(Message mimeMessage, MessageFullViewFactory.MetaDataWithContent message) {
-            return Optional.ofNullable(mimeMessage.getDate())
-                .map(Date::toInstant)
-                .orElse(message.getInternalDate());
-        }
-
-
-        static Message parse(MessageResult message) throws MailboxException {
-            try {
-                return Message.Builder
-                    .of()
-                    .use(MimeConfig.PERMISSIVE)
-                    .parse(message.getFullContent().getInputStream())
-                    .build();
-            } catch (IOException e) {
-                throw new MailboxException("Unable to parse message: " + e.getMessage(), e);
-            }
-        }
-
-
-        static Message parse(MessageFullViewFactory.MetaDataWithContent message) throws MailboxException {
-            try {
-                return Message.Builder
-                        .of()
-                        .use(MimeConfig.PERMISSIVE)
-                        .parse(message.getContent())
-                        .build();
-            } catch (IOException e) {
-                throw new MailboxException("Unable to parse message: " + e.getMessage(), e);
-            }
+        static Message parse(InputStream messageContent) throws IOException {
```

Comment:
I tried to do the same in #2926 , just saying it was rejected with this as a reason : 

> I don't like that method because it hides a builder.
> 
> Mime4J parsing is not that complex that it should be abstracted away.
> 
> Duplication of that method is acceptable to me.

Reply:
Here I think that it is OK as it is bounded to the context of Message view factories...